Sources/Basic/TerminalController.swift

```diff
@@ -0,0 +1,252 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+import func POSIX.getenv
+
+private protocol OutputFileStreamProtocol: OutputStream {
+    var fd: Int32 { get }
+    var file: UnsafeMutablePointer<FILE> { get }
+}
+
+final public class OutputFileStream: OutputFileStreamProtocol {
+    private let fd: Int32
+    private let file: UnsafeMutablePointer<FILE>
+
+    init(file: UnsafeMutablePointer<FILE>) {
+        self.file = file
+        self.fd = fileno(file)
+    }
+
+    public func write(_ string: String) {
+        _ = fputs(string, file)
+    }
+}
+
+public var stdout = OutputFileStream(file: libc.stdout)
+public var stderr = OutputFileStream(file: libc.stderr)
+
+/// Terminal color choices.
+public enum TermColor {
+    case noColor
+
+    case red
+    case green
+    case yellow
+
+    case white
+    case black
+    case grey
+
+    /// Returns the color code which can be prefixed on a string to display it in that color.
+    /// Note: To avoid unexpected behavior the color codes are private and not exposed to clients
+    /// other than TerminalController.
```

Comment:
Err that comment came out wrong, I meant to write there is no need for clients to know about this string because they shouldn't be manually inserting the color code in string. I'll reword